qiskit/circuit/random/utils.py

```diff
@@ -11,16 +11,385 @@
+    random based on the probability attached to the qubit-pair, the 2Q gates are applied on the
+    qubit-pairs which are idle for that particular iteration, this is to make sure that for a particular
+    iteration only one circuit layer exists, now, if `insert_1q_oper` is set to True, randomly
+    chosen 1Q gates are applied to qubits that are still idle for that particular iteration, after
+    applying 2Q gates for that particular iteration.
+
+    Example:
+
+    .. plot::
+       :alt: Pass in interaction graph and minimum 2Q gate per edge as a bare minimum.
+       :include-source:
+
+       from qiskit.circuit.random.utils import random_circuit_from_graph
+       import rustworkx as rx
+       pydi_graph = rx.PyDiGraph()
+       pydi_graph.add_nodes_from(range(7))
+       cp_map = [(0, 1, 0.18), (1, 2, 0.15), (2, 3, 0.15), (3, 4, 0.22), (4, 5, 0.13), (5, 6, 0.17)]
+       pydi_graph.add_edges_from(cp_map)
+       qc = random_circuit_from_graph(pydi_graph, min_2q_gate_per_edge=1, measure = True)
+       qc.draw(output='mpl')
+
+    Args:
+        interaction_graph (PyGraph | PyDiGraph): Interaction Graph
+        min_2q_gate_per_edge (int): Minimum number of times every qubit-pair must be used
+        in the random circuit.
+        max_operands (int): maximum qubit operands of each gate(should be 1 or 2)
+        (optional, default:2)
+        measure (bool): if True, measure all qubits at the end. (optional, default: False)
+        conditional (bool): if True, insert middle measurements and conditionals.
+        (optional, default: False)
+        reset (bool): if True, insert middle resets. (optional, default: False)
+        seed (int): sets random seed. (If `None`, a random seed is chosen) (optional)
+        insert_1q_oper (bool): Insert 1Q operations to the circuit. (optional, default: True)
+        prob_conditional (float): Probability less than 1.0, this is used to control the occurence
+        of conditionals in the circuit. (optional, default: 0.1)
+        prob_reset (float): Probability less than 1.0, this is used to control the occurence of
+        reset in the cirucit. (optional, default: 0.1)
+
+    Returns:
+        QuantumCircuit: constructed circuit
+
+    Raises:
+        CircuitError: When `max_operands` is not 1 or 2.
+        CircuitError: When `max_operands` is set to 1, but no 1Q operations are allowed by setting
+        `insert_1q_oper` to false.
+        CircuitError: When the interaction graph has no edges, so only 1Q gates are possible in
+        the circuit, but `insert_1q_oper` is set to False.
+        ValueError: when any edge have probability None but not all or, any of the probabilities
+        are negatve.
+    """
+
+    # max_operands should be 1 or 2
+    if max_operands not in {1, 2}:
+        raise CircuitError(
+            "This function is intended to only work on single-qubit and two-qubit gates"
```

Comment:
```suggestion
            "`max_operands` should be either 1 or 2"
```
I think this is a clearer message, mentioning the argument explicitly